packages/core/src/tools/catchUserErrors.ts

```diff
@@ -0,0 +1,11 @@
+export function catchUserErrors<Args extends any[], R>(fn: (...args: Args) => R, errorMsg: string) {
+  return (...args: Args) => {
+    let result
+    try {
+      result = fn(...args)
+    } catch (err) {
+      console.error(errorMsg, err)
+    }
+    return result
```

Comment:
```suggestion
    try {
      return fn(...args)
    } catch (err) {
      console.error(errorMsg, err)
    }
```
do we need `result`?